src/box/lua/config/instance_config.lua

```diff
@@ -2360,6 +2360,12 @@ return schema.new('instance_config', schema.record({
         }, {
             default = 'new',
         }),
+        box_cfg_replication_synchro_timeout = schema.enum({
+            'old',
+            'new',
+        }, {
+            default = 'old',
+        }),
```

Comment:
Nit: I would add new compat options to the end of the list, so we have options from older releases first and then newer ones. As I see, the options from 3.x (which all has `old` default) are at the end.